manager/integration/tests/test_basic.py

```diff
@@ -600,6 +600,45 @@ def backup_failure_predicate(b):
     cleanup_volume(client, volume)
 
 
+@pytest.mark.skip(reason="TODO")
+def test_backup_block_deletion():  # NOQA
+    """
+    Test backup block deletion
+
+    Context:
+
+    We want to make sure that we only delete non referenced backup blocks,
+    we also don't want to delete blocks while there other backups in progress.
+    The reason for this is that we don't yet know which blocks are required by
+    the in progress backup, so blocks deletion could lead to a faulty backup.
+
+    Setup:
+
+    1. Create a volume and attach to the current node
+    2. Setup NFS backupstore since we can manipulate the content easily
+
+    Steps:
+
+    1.  Write 2 backup blocks of data on to the volume
+    2.  Create backup(1) of the volume
+    3.  Overwrite the first of the backup blocks of data on the volume
+    4.  Create backup(2) of the volume
+    5.  Overwrite the first of the backup blocks of data on the volume
+    6.  Create backup(3) of the volume
+    7.  Verify backup block count == 4
```

Comment:
Also check the backup directory, search for `.blk` make sure it matched the block count.

Reply:
For the block count check one wants to check the volume.cfg as well as the blocks on disk, like I do in the engine integration test. Should I include these details in the descriptions? since they are implementation details?

Reply:
Yes, include them when encountered for the first time.